File: clients/java-okhttp-moshi/test_service/clients/echo/EchoSuccessResponse.java
package test_service.clients.echo;

import test_service.models.*;

public interface EchoSuccessResponse {
	class Ok implements EchoSuccessResponse {
		public OkResult body;

		public Ok() {
		}

		public Ok(OkResult body) {
			this.body = body;
		}
	}

	class Created implements EchoSuccessResponse {
		public CreatedResult body;

		public Created() {
		}

		public Created(CreatedResult body) {
			this.body = body;
		}
	}

	class Accepted implements EchoSuccessResponse {
		public AcceptedResult body;

		public Accepted() {
		}

		public Accepted(AcceptedResult body) {
			this.body = body;
		}
	}
}

File: clients/kotlin-okhttp-jackson/test_service/clients/echo/EchoSuccessResponse.kt
package test_service.clients.echo

import test_service.models.*

interface EchoSuccessResponse {
	class Ok(var body: OkResult) : EchoSuccessResponse

	class Created(var body: CreatedResult) : EchoSuccessResponse

	class Accepted(var body: AcceptedResult) : EchoSuccessResponse
}
